app/nexus/internal/route/secret_get.go

```diff
@@ -85,9 +86,13 @@ func routeGetSecret(
 	version := request.Version
 	path := request.Path
 
-	secret, exists := state.GetSecret(path, version)
-	if !exists {
-		log.Log().Info("routeGetSecret", "msg", "Secret not found")
+	secret, err := state.GetSecret(path, version)
+	if err != nil {
+		if errors.Is(err, store.ErrSecretNotFound) {
+			log.Log().Warn("routeGetSecret", "msg", err.Error())
+		} else if errors.Is(err, store.ErrSecretSoftDeleted) {
+			log.Log().Warn("routeGetSecret", "msg", err.Error())
+		}
```

Comment:
We are logging the same thing; this comparison don't seem to add any value.

Consider using `errors.Join()` to add additional context to the message.

---

Also `.Warn()` might be too harsh for a 404-like "secret not found" error; `log.Info()` might be better.